extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -190,17 +190,37 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async showPreviewFile(urlToOpen?: string): Promise<void> {
 		if (this.currentBook) {
-			const bookRoot = this.currentBook.bookItems[0];
-			const sectionToOpen = bookRoot.findChildSection(urlToOpen);
-			const urlPath = sectionToOpen ? sectionToOpen.url : bookRoot.tableOfContents.sections[0].url;
-			const sectionToOpenMarkdown: string = path.join(this.currentBook.bookPath, Content, urlPath.concat('.md'));
-			// The Notebook editor expects a posix path for the resource (it will still resolve to the correct fsPath based on OS)
-			const sectionToOpenNotebook: string = path.posix.join(this.currentBook.bookPath, Content, urlPath.concat('.ipynb'));
-			if (await fs.pathExists(sectionToOpenMarkdown)) {
-				this.openMarkdown(sectionToOpenMarkdown);
+			let urlPath: string;
+			if (this.currentBook.isNotebook) {
+				urlPath = urlToOpen && this.currentBook.bookPath === urlToOpen ? this.currentBook.bookPath : undefined;
```

Comment:
When would this be true? Is this for the case of opening a regular folder of Notebooks? Why wouldn't you default to the "first" Notebook in the folder then - similar to what we do with Books? I guess I'm not exactly sure when this is called so maybe it makes sense not to open a file immediately.

Reply:
So we call this from OpenBook for every notebook. In case of a Book , it has items and we open the first one when no urlToOpen is given. In case of folder with notebooks, this get's called for every notebook with urlToOpen and if we default like books, we end up opening every notebook in the folder. Hence opening only the passed urlToOpen (if there is one or none at all).